internal/controller/dnsrecord_controller.go

```diff
@@ -193,6 +192,41 @@ func (r *DNSRecordReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		ctx, logger = r.setLogger(ctx, baseLogger, dnsRecord)
 	}
 
+	// Ensure we have provider secret
+	if !dnsRecord.HasProviderSecretAssigned() {
+		if dnsRecord.Spec.ProviderRef.Name != "" {
+			dnsRecord.Status.ProviderRef = dnsRecord.Spec.ProviderRef
+		} else {
+			// try to find the default secret
+			defaultSecretList := &v1.SecretList{}
+			err = r.Client.List(ctx, defaultSecretList, client.InNamespace(dnsRecord.Namespace), client.HasLabels{v1alpha1.DefaultProviderSecretLabel})
```

Comment:
Should this check for the label having a "truthy" value? Assume "kuadrant.io/default-provider=false" would also work here.